### ISSUE TITLE
ImportError: cannot import name 'ugettext_lazy' from 'django.utils.translation' on Django latest version

### ISSUE THREAD

**Issue by nalbert9** [graphql-python/graphene-django#1424]
Hi,
I am writing to report an issue I encountered while using the graphql-python/graphene-django package in a Django 4.2 project. When attempting to import the schema module, I encountered the following ImportError:

**ImportError: cannot import name 'ugettext_lazy' from 'django.utils.translation' (/lib/python3.10/site-packages/django/utils/translation/__init__.py)**

Upon investigating, it appears that this issue arises due to the deprecation of the **ugettext_lazy** function in **Django 4.2**, which has been replaced with **gettext_lazy.** Could you please assist in resolving this import error? I would greatly appreciate any guidance or potential workarounds you can provide.

Regards,
Badji

**Comment by ulgens** [comment]
@nalbert9 Which version of graphene-django you are using?

**Comment by nalbert9** [comment]
Hi @ulgens 
graphene-django Version: 3.1.1 and also Version: 2.16.0

**Comment by sjdemartini** [comment]
Are you sure you're using graphene-django 3.1.1? Would you be able to share a full stack trace and point to the line of code in graphene-django that has the error? 

`ugettext_lazy` was replaced with `gettext_lazy` already in https://github.com/graphql-python/graphene-django/commit/dd0d6ef28ffce33d30c4c8908f48a9d44a2b75ce, which has been published since v3.0.0b1. As you can see in a GH code search, [`ugettext_lazy` is not present anywhere in the repo](https://github.com/search?q=repo%3Agraphql-python%2Fgraphene-django%20ugettext_lazy&type=code), while [`gettext_lazy` is](https://github.com/search?q=repo%3Agraphql-python%2Fgraphene-django%20gettext_lazy&type=code)

**Comment by nalbert9** [comment]
Hi @sjdemartini 
Thank you for your response. You will find my configuration below:

❯ pip show graphene-django
Name: graphene-django
Version: 3.1.1
Summary: Graphene Django integration
Home-page: https://github.com/graphql-python/graphene-django
Author: Syrus Akbary
Author-email: me@syrusakbary.com
License: MIT
Location: ../anaconda3/lib/python3.10/site-packages
Requires: Django, graphene, graphql-core, graphql-relay, promise, text-unidecode
Required-by: django-graphql-auth, django-graphql-jwt

ImportError: Could not import 'customers.schema.schema' for Graphene setting 'SCHEMA'. ImportError: cannot import name 'ugettext' from 'django.utils.translation' (../anaconda3/lib/python3.10/site-packages/django/utils/translation/__init__.py).
[11/Jun/2023 19:51:47] "GET /graphql/ HTTP/1.1" 500 158811

**Comment by sjdemartini** [comment]
@nalbert9 is there any more detail than the import error above? Do you have a complete stack trace that shows what specific file or code is trying to import `ugetttext`? It seems the problem is likely coming from some other library or code you are using, as graphene-django 3.1.1 does not contain any usage of `ugettext`.

**Comment by nalbert9** [comment]
@sjdemartini, I don't have ugetttext in the code.  You will find the trackback:
 May be related to **from graphql_jwt.decorators import login_required**?

Environment:

Request Method: GET
Request URL: http://127.0.0.1:8000/graphql/

Django Version: 4.2.2
Python Version: 3.10.9
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.sites',
 'customers',
 'graphene_django',
 'allauth',
 'allauth.account',
 'allauth.socialaccount',
 'allauth.socialaccount.providers.google',
 'graphql_auth',
 'corsheaders']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'corsheaders.middleware.CorsMiddleware']



Traceback (most recent call last):
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/graphene_django/settings.py", line 77, in import_from_string
    module = importlib.import_module(module_path)
  File "/Users/badji/anaconda3/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
    <source code not available>
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
    <source code not available>
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
    <source code not available>
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
    <source code not available>
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
    <source code not available>
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
    <source code not available>
  File "/Users/badji/Projects/users/customers/schema.py", line 4, in <module>
    from graphql_jwt.decorators import login_required
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/graphql_jwt/__init__.py", line 1, in <module>
    from . import relay
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/graphql_jwt/relay.py", line 5, in <module>
    from . import mixins
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/graphql_jwt/mixins.py", line 1, in <module>
    from django.utils.translation import ugettext as _

During handling of the above exception (cannot import name 'ugettext' from 'django.utils.translation' (/Users/badji/anaconda3/lib/python3.10/site-packages/django/utils/translation/__init__.py)), another exception occurred:
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 56, in wrapper_view
    return view_func(*args, **kwargs)
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/django/views/generic/base.py", line 97, in view
    self = cls(**initkwargs)
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/graphene_django/views.py", line 103, in __init__
    schema = graphene_settings.SCHEMA
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/graphene_django/settings.py", line 124, in __getattr__
    val = perform_import(val, attr)
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/graphene_django/settings.py", line 63, in perform_import
    return import_from_string(val, setting_name)
  File "/Users/badji/anaconda3/lib/python3.10/site-packages/graphene_django/settings.py", line 86, in import_from_string
    raise ImportError(msg)

Exception Type: ImportError at /graphql/
Exception Value: Could not import 'customers.schema.schema' for Graphene setting 'SCHEMA'. ImportError: cannot import name 'ugettext' from 'django.utils.translation' (/Users/badji/anaconda3/lib/python3.10/site-packages/django/utils/translation/__init__.py).

**Comment by sjdemartini** [comment]
@nalbert9 thanks for sharing that full traceback. Yes, it seems that's correct, the problematic code appears to be from `graphql_jwt`, as it shows in the traceback:

```
File "/Users/badji/anaconda3/lib/python3.10/site-packages/graphql_jwt/mixins.py", line 1, in
from django.utils.translation import ugettext as _
```

For what it's worth, they claim to have fixed this in version 0.3.1 [according to their changelog](https://github.com/flavors/django-graphql-jwt/blob/67ab53d1bfeba908a01f22ce39f2486af6fa7dd2/CHANGELOG.rst#031) via https://github.com/flavors/django-graphql-jwt/pull/160. You should try upgrading to the latest version of that library.

Closing this out, as it's not due to `graphene-django`.
